react-app/src/pages/WeatherPage.tsx

```diff
@@ -1,44 +1,79 @@
 import axios from "axios";
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
 import Container from "../components/Container";
 import DailyWeather from "../components/DailyWeather";
-import {Weather} from "../interfaces/Weather.interface";
+import { Weather, ICity } from "../interfaces/Weather.interface";
 import "../styles/Custom.module.css";
 
 const WeatherPage: React.FC = () => {
   const [weather, setWeather] = useState<Weather | null>(null);
+  const [cityData, setCityData] = useState<ICity | null>(null);
+  const [error, setError] = useState<boolean>(false);
```

Comment:
What is the purpose of this `error` state? I don't see any application of it anywhere -- but I might be blind!

Reply:
I did a conditional rendering in the html render section based on the `error` state.